core/src/main/java/com/netease/arctic/op/ArcticTableOperations.java

```diff
@@ -65,13 +71,67 @@ public String metadataFileLocation(String fileName) {
     return ops.metadataFileLocation(fileName);
   }
 
+  private String metadataFileLocation(TableMetadata metadata, String filename) {
+    String metadataLocation = metadata.properties().get(TableProperties.WRITE_METADATA_LOCATION);
+
+    if (metadataLocation != null) {
+      return String.format("%s/%s", LocationUtil.stripTrailingSlash(metadataLocation), filename);
+    } else {
+      return String.format("%s/%s/%s", metadata.location(), METADATA_FOLDER_NAME, filename);
+    }
+  }
+
   @Override
   public LocationProvider locationProvider() {
     return ops.locationProvider();
   }
 
   @Override
   public TableOperations temp(TableMetadata uncommittedMetadata) {
-    return ops.temp(uncommittedMetadata);
+    return new TableOperations() {
+      @Override
+      public TableMetadata current() {
+        return uncommittedMetadata;
+      }
+
+      @Override
+      public TableMetadata refresh() {
+        throw new UnsupportedOperationException(
+            "Cannot call refresh on temporary table operations");
+      }
+
+      @Override
+      public void commit(TableMetadata base, TableMetadata metadata) {
+        throw new UnsupportedOperationException("Cannot call commit on temporary table operations");
+      }
+
+      @Override
+      public String metadataFileLocation(String fileName) {
+        return ArcticTableOperations.this.metadataFileLocation(
```

Comment:
Maybe ops.metadataFileLocation(String fileName)

Reply:
This operation is for `uncommittedMetadata`, I think it should be consistent with `org.apache.iceberg.BaseMetastoreTableOperations#temp`.
Although I'm not sure what the difference is between the two method.

Reply:
We don't know  which implement of TableOperations ops is, so we should rewrite the temp() as :
![image](https://user-images.githubusercontent.com/26699250/232950540-972ac3ef-43ca-446b-ac72-abf041e5b4de.png)

Reply:
I see.